lib/adapters/xhr.js

```diff
@@ -168,6 +168,21 @@ module.exports = function xhrAdapter(config) {
       request.upload.addEventListener('progress', config.onUploadProgress);
     }
 
+    // Check maxContentLength if needed
+    if (config.maxContentLength > 0) {
+      request.addEventListener('progress', function checkMaxContentLength(event) {
```

Comment:
There is a note at https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest saying "You need to add the event listeners before calling open() on the request". I also found https://stackoverflow.com/a/57653034/2881634 warning on this. The file already seems to have other event listeners after open, so this is not the first time it is done wrong (if there really are any browsers where adding event listeners after open does not work). Perhaps this would be a good time to fix the event listeners to be added before open, just in case.